qbittorrent/client.py

```diff
@@ -159,53 +172,74 @@ def torrents(self, **filters):
         :param offset: Set offset (if less than 0, offset from end).
 
         :return: list() of torrent with matching filter.
+        For example: qb.torrents(filter='downloading', sort='ratio').
         """
         params = {}
         for name, value in filters.items():
             # make sure that old 'status' argument still works
             name = 'filter' if name == 'status' else name
             params[name] = value
 
-        return self._get('query/torrents', params=params)
+        return self._get('torrents/info', params=params)
 
     def get_torrent(self, infohash):
         """
         Get details of the torrent.
 
         :param infohash: INFO HASH of the torrent.
         """
-        return self._get('query/propertiesGeneral/' + infohash.lower())
+        return self._get('torrents/properties?hash=' + infohash.lower())
 
     def get_torrent_trackers(self, infohash):
         """
         Get trackers for the torrent.
 
         :param infohash: INFO HASH of the torrent.
         """
-        return self._get('query/propertiesTrackers/' + infohash.lower())
+        return self._get('torrents/trackers?hash=' + infohash.lower())
 
     def get_torrent_webseeds(self, infohash):
         """
         Get webseeds for the torrent.
 
         :param infohash: INFO HASH of the torrent.
         """
-        return self._get('query/propertiesWebSeeds/' + infohash.lower())
+        return self._get('torrents/webseeds?hash=' + infohash.lower())
 
     def get_torrent_files(self, infohash):
         """
         Get list of files for the torrent.
 
         :param infohash: INFO HASH of the torrent.
         """
-        return self._get('query/propertiesFiles/' + infohash.lower())
+        return self._get('torrents/files?hash=' + infohash.lower())
+
+    def get_torrent_piece_states(self, infohash):
+        """
+        Get list of all pieces (in order) of a specific torrent.
+
+        :param infohash: INFO HASH of the torrent.
+        :return: array of states (integers).
+        """
+        return self._get('torrents/pieceStates?hash=' + infohash.lower())
+
+    def get_torrent_piece_hashes(self, infohash):
+        """
+        Get list of all hashes (in order) of a specific torrent.
+
+        :param infohash: INFO HASH of the torrent.
+        :return: array of hashes (strings).
+        """
+        return self._get('torrents/pieceHashes?hash=' + infohash.lower())
+
 
     @property
     def global_transfer_info(self):
         """
-        Get JSON data of the global transfer info of qBittorrent.
+        :return: dict{} of the global transfer info of qBittorrent.
+
         """
-        return self._get('query/transferInfo')
+        return self._get('transfer/info')
 
     @property
     def preferences(self):
```

Comment:
Preferences isn't working due to the way it is implemented. The fix for this is to replace

```python
super(Proxy, self).__init__(url)
```

with 

```python
self.url = url
```

Reply:
Fixed in 20f80c8.